Mathlib/Logic/Equiv/Defs.lean

```diff
@@ -625,6 +625,14 @@ is equivalent to `β a`. -/
 #align equiv.Pi_subsingleton_symm_apply Equiv.piSubsingleton_symm_apply
 #align equiv.Pi_subsingleton Equiv.piSubsingleton
 
+/-- The equivalence `(∀ i, β i) ≃ β ⋆` when the domain of `β` only contains `⋆` -/
+@[simps (config := .asFn)]
+def piUnique [Unique α] (β : α → Sort*) : (∀ i, β i) ≃ β default where
+  toFun f := f default
+  invFun := uniqueElim
+  left_inv f := by ext i; cases Unique.eq_default i; rfl
+  right_inv x := rfl
```

Comment:
Maybe:
```suggestion
abbrev piUnique [Unique α] (β : α → Sort*) : (∀ i, β i) ≃ β default := piSubsingleton
```

Reply:
I have removed `piSubsingleton` and now use exclusively use `piUnique`.
* There was only one use of `piSubsingleton` in mathlib, which is on a type that has the `Unique` class. 
* I think that stating it for `Unique` is more natural than for a subsingleton type with an element. 
* `piSubsingleton_symm_apply` was bad, since it would unfold the equivalence to a very ugly expression with casts. (for `piUnique` that is hidden behind the function `uniqueElim`).

Reply:
I guess the existence of `uniqueElim` and the absense of `subsingletonElim i` is a convincing enough argument for why `piUnique` is overkill.

Reply:
It turns out that I was the one who asked for this in the first place, in https://github.com/leanprover-community/mathlib/pull/12040#discussion_r806349297. The original motivation was to avoid a `convert`, but it seems you don't have one of those any more anyway.

Reply:
It turned out to require still some changes in the `Sylow` file, but I don't think they are deteriorations: we had a `Inhabited` and `Subsingleton` instances on `Sylow G p` (under conditions), so it makes sense to upgrade this to a `Unique` instance.